Addition of Multiple Channels

kirtana-poikayil opened the pull request:
Features:

1. Default is Choose Stream and Choose Channels.
2. Plot 4 or less stream-channel combinations.
3. In order for the stream-channel to be plotted, user has to click start in the inventory area. Then they have to select the started stream and the channel in the stream. The visuals should pop-up with a legend on the first graph. This legend applies to all 3 graphs.
4. In order to remove a stream-channel combination from the graph, click Choose Stream or Choose Channel. 
5. If a user clicks stop for a stream in inventory but doesn't remove it, that particular set of data is paused on the graph while other stream-channels will continue plotting.
6. Also, if user goes to band settings, there is now an option to choose which bands to plot. (Ex: User can choose to hide Alpha Band)

Also, happy holidays!

roenzen608-design reviewed (approved):
Looks good! Works perfectly with a multi-channel sim and being able to choose up to 4 different channels of data is nice. 

Adding a second sim stream also has exposed a visual bug with the inventory that we should try to fix in the future where starting two streams makes it so only one stream is viewable in the inventory at a time, but it doesn't affect us being able to interact with the object in the inventory. Specifically, the labels of "start/stop" and the stream name become invisible. This isn't something caused by your code though (I don't think).

Good work and thank you! I should've gotten the simulated two channel stream to you earlier but it slipped my mind.

zbalkhy commented on src/plotterView.py:
```diff
@@ -105,171 +128,90 @@ def _setup_controls(self):
+        stream_name = stream_var.get()
+        
+        # If default "Choose Stream" is selected, reset channel
+        if stream_name == "Choose Stream":
```
can replace the "Choose Stream" here with the variable CHOOSE_STREAM

zbalkhy commented on src/plotterView.py:
```diff
@@ -105,171 +128,90 @@ def _setup_controls(self):
+        
+        # If default "Choose Stream" is selected, reset channel
+        if stream_name == "Choose Stream":
+            channel_var.set("Choose Channel")
```
can replace "Choose Channel" with CHOOSE_CHANNEL variable

zbalkhy commented on src/plotterView.py:
```diff
@@ -105,171 +128,90 @@ def _setup_controls(self):
+            return
+
+        n_ch = data.shape[1]
+        channel_choices = ["Choose Channel"] + [str(i) for i in range(1, n_ch + 1)]
```
Same comment here with "Choose Channel" and wherever else it appears

kirtana-poikayil commented:
Hi,
I've added MAX_CHANNELS and fixed the places where I needed to add CHOOSE_CHANNEL and CHOOSE_STREAM. Is it alright if I merge?

Thanks for reviewing my code!
